### DISCUSSION TITLE
ViewState and BindableState

### DISCUSSION THREAD

**Discussion by JaapWijnen** [pointfreeco/swift-composable-architecture#1397] [Q&A] [answered]
Is it possible to combine scoping state to a ViewState in a view whilst still being able to use the super useful @BinableState wrappers? I haven't been able to make them play well together.

**Comment by tgrapperon** [comment, marked as answer]
Hey @JaapWijnen, this is indeed cumbersome right now (#769). [There are recent experiments](https://github.com/pointfreeco/swift-composable-architecture/tree/protocol-bindable-view-state) to improve the ergonomics.

**Reply by JaapWijnen** [reply]
thanks!
